src/main/resources/JavadocLexer.g4

```diff
@@ -73,13 +74,12 @@ import org.antlr.v4.runtime.Token;
     @Override
     public void emit(Token token) {
         super.emit(token);
-        if (token.getType() == TAG_NAME) {
+        if (token.getType() == TAG_NAME && !LexerUtility.isSelfClosing(_input)) {
```

Comment:
It is not clear to me why we want to iterate through every tag we have for every single token that we lex, regardless of type. The lexer is the most performance sensitive part of our project; we should avoid any extra work whenever possible. This is why I chose to just dump everything into the collections indiscriminately, then lazily build the collection of unclosed tags only when we need it.

Reply:
Let's add this entire condition to the utility